poly-synth/src/poly-synth.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @class PolySynth
+ * @description Manages the life cycle of voices.
+ */
+class PolySynth {
+  /**
+   * @constructor
+   * @param {AudioContext} context The audio context.
+   */
+  constructor(context) {
+    if (!(context instanceof AudioContext))
+      throw context + ' is not a valid audio context.';
+
+    this.context_ = context;
+    this.activeVoices_ = [];
+    this.releasedVoices_ = [];
+
+    // The default maximum cutoff is set below the upper edge of human
+    // hearing and the minimum cutoff is above the lower edge of human hearing.
+    // Frequencies at either extreme are not typically used in electronic music.
+    this.lowPassMaxCutoff = 12000;
+    this.lowPassMinCutoff = 60;
+
+    // The meaning of each parameter is defined in |getParameters()|.
+    this.parameters_ = {
+      cutoff: this.lowPassMaxCutoff,
+      attack: 0,
+      decay: 0,
+      sustain: 0.1,
+      release: 0
+    };
+
+    // The client is responsible for connecting |this.output| to a destination.
+    this.output = new GainNode(this.context_);
+  }
+
+  /**
+   * @typedef {Object} Parameters Parameters which affect the output of a voice
+   *                              if set before the voice is constructed.
+   * @property {Number} attack Seconds until full amplitude.
+   * @property {Number} decay Seconds until sustain level.
+   * @property {Number} sustain The steady amplitude of the note as it
+   *                            is pressed.
+   * @property {Number} release Seconds between the release of a note
+   *                            and zero amplitude.
+   * @property {Number} release Seconds between the release of a note
+   *                            and zero amplitude.
+   */
+  /**
```

Comment:
You can add an empty line to clarify the structure of comment and code. Also I am not sure what this comment is for. Seems redundant.